lua/agrp.lua

```diff
@@ -1,75 +1,64 @@
-local M = {
-  funcs = {},
-  my_name = (function()
-    local file = debug.getinfo(1, 'S').source
-    return file:match'/(%a+)%.lua$'
-  end)()
-}
+local M = {}
 
-local function make_command(events, patterns, ...)
-  local args = {...}
-  local options, cmd_or_func
-  if #args == 2 then
-    options = args[1]
-    cmd_or_func = args[2]
-  else
-    options = {}
-    cmd_or_func = args[1]
-  end
-  local opts = ''
-  for _, o in ipairs(options) do
-    opts = ('%s ++%s'):format(opts, o)
-  end
-  local command
-  if type(cmd_or_func) == 'string' then
-    command = cmd_or_func
-  else
-    table.insert(M.funcs, cmd_or_func)
-    command = ([[lua require'%s'.funcs[%d]()]]):format(M.my_name, #M.funcs)
-  end
-  return ('autocmd %s %s%s %s'):format(events, patterns, opts, command)
+local function is_vim_func_string(s)
+  return type(s) == "string" and s:match "^[bwtglsav]:[_%d%w]+$"
 end
 
-local function manage_definitions(cmds, definitions)
+local function manage_definitions(definitions, group)
   for key, definition in pairs(definitions) do
-    vim.validate{
-      definition = {definition, 'table'},
+    vim.validate {
+      definition = { definition, "table" },
     }
-    if type(key) == 'number' then
-      if #definition == 3 or #definition == 4 then
-        table.insert(cmds, make_command(unpack(definition)))
+    local opt = { group = group }
+    local event
+    local cb_or_cmd
+    if type(key) == "number" then
+      if #definition == 3 then
+        opt.pattern = definition[2]
+        event = definition[1]
+        cb_or_cmd = definition[3]
+      elseif #definition == 4 then
+        opt.once = definition[1].once and true or false
+        opt.nested = definition[1].nested and true or false
+        opt.pattern = definition[3]
+        event = definition[2]
+        cb_or_cmd = definition[4]
       else
-        error'each definition should have 3 values (+options (once, nested))'
+        error "each definition should have 3 values (+options (once, nested))"
       end
     else
       for _, d in ipairs(definition) do
```

Comment:
in this loop, each definition `d` is not being registered as an autocmd only the last one.

Reply:
Thank you for your review! I clean up logic to call `nvim_create_autocmd()`.